cmd/root/root.go

```diff
@@ -151,5 +161,40 @@ Stack Trace:
 		}
 	}
 
+	exitCode := 0
+	if err != nil {
+		exitCode = 1
+	}
+
+	uploadTelemetry(cmd.Context(), commandString(cmd), startTime, exitCode)
 	return err
 }
+
+func uploadTelemetry(ctx context.Context, cmdStr string, startTime time.Time, exitCode int) {
+	// Return early if there are no logs to upload.
+	if !telemetry.HasLogs(ctx) {
+		log.Debugf(ctx, "no telemetry logs to upload")
+		return
+	}
+
+	// Telemetry is disabled. We don't upload logs.
+	if env.Get(ctx, telemetry.DisableEnvVar) != "" {
+		log.Debugf(ctx, "telemetry upload is disabled. Not uploading any logs.")
+		return
+	}
+
+	telemetry.SetExecutionContext(ctx, protos.ExecutionContext{
+		CmdExecID:       cmdExecId,
+		Version:         build.GetInfo().Version,
+		Command:         cmdStr,
+		OperatingSystem: runtime.GOOS,
+		DbrVersion:      env.Get(ctx, dbr.EnvVarName),
+		ExecutionTimeMs: time.Since(startTime).Milliseconds(),
+		ExitCode:        int64(exitCode),
+	})
+
+	err := telemetry.Upload(ctx, ConfigUsed(ctx))
```

Comment:
Why not hide SetExecutionContext/HasLogs/env var checking inside telemetry packages? So the code would look like this:

`telemetry.Upload(ctx, ... ExecutionContext{...}) // updates logs with execution context, checks env var if upload is needed`

Reply:
Moved `SetExecutionContext` inside and the env var code inside. We cannot move `HasLogs` because not every command would configure a config in the context, and for those cases `ConfigUsed(ctx))` would panic.

Reply:
I can work on a followup PR that consolidates `ConfigUsed` and other clients in the `libs/command` library. Ref: https://github.com/databricks/cli/pull/2439

Reply:
Once https://github.com/databricks/cli/pull/2440 is merged, we can get rid of HasLogs as well.

Reply:
Removed the HasLogs function.